tests/Unit/Validator/Constraints/GiftCardIsNotExpiredValidatorTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Setono\SyliusGiftCardPlugin\Unit\Validator\Constraints;
+
+use PHPUnit\Framework\TestCase;
+use Prophecy\PhpUnit\ProphecyTrait;
+use Setono\SyliusGiftCardPlugin\Model\GiftCardInterface;
+use Setono\SyliusGiftCardPlugin\Validator\Constraints\GiftCardIsNotExpired;
+use Setono\SyliusGiftCardPlugin\Validator\Constraints\GiftCardIsNotExpiredValidator;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Violation\ConstraintViolationBuilderInterface;
+
+final class GiftCardIsNotExpiredValidatorTest extends TestCase
+{
+    use ProphecyTrait;
+
+    /**
+     * @test
+     */
+    public function it_throws_exception_if_constraint_has_wrong_type(): void
+    {
+        $validator = new GiftCardIsNotExpiredValidator();
+        $constraint = new NotBlank();
+
+        $this->expectExceptionObject(new UnexpectedTypeException($constraint, GiftCardIsNotExpired::class));
+        $validator->validate('super', $constraint);
+    }
+
+    /**
+     * TODO: see how we can test this
```

Comment:
I don't know if it helps, but usually you can get a lot of good info from looking a Symfony test cases. Here's an example: https://github.com/symfony/validator/blob/5.4/Tests/Constraints/EmailValidatorTest.php

Reply:
I wanted to test the form itself, not the validation. But I've never seen any test about form :thinking:

Reply:
This you mean? https://symfony.com/doc/current/form/unit_testing.html

Reply:
:heart: thanks